docs/2.guide/4.recipes/4.sessions-and-authentication.md

```diff
@@ -0,0 +1,403 @@
+
+Here's an example registration form for reference:
+
+```vue [pages/register.vue]
+<script setup lang="ts">
+  const form = ref({
+    name: "",
+    email: "",
+    password: "",
+  });
+
+  const error = ref(null);
+
+  async function submitForm() {
+    // clear any previous errors
+    error.value = null;
+
+    // perform the login
+    try {
+      await $fetch("/api/register", {
+        method: "POST",
+        body: form.value,
+      });
+    } catch (e) {
+      // if there's an error, set the error message and return early
+      error.value = "Error Registering User";
+      return;
+    }
+    // Refresh the session status now that the user is logged in
+    // This is a fetch() function from auth-utils to refresh session data in the client, and is not the same as $fetch
+    const { fetch } = useUserSession();
+    await fetch();
+    // you may want to use something like Pinia to manage global state of the logged-in user
+    // update Pinia state here...
+
+    // take the user to the auth-only users index page now that they're logged in
+    await navigateTo("/users");
+
+    // Alternative - Don't use Nuxt Router here so that we can easily trigger a whole page load and get the whole UI refreshed now that the user is logged in.
+    // window.location.href = "/users";
+  }
+</script>
+
+<template>
+  <div>
+    <form class="space-y-2" @submit.prevent="submitForm">
+      <div>
+        <label for="name" class="block text-sm uppercase">Name</label>
+
+        <input id="name" name="name" v-model="form.name" class="rounded-md px-2 py-1" required />
+      </div>
+
+      <div>
+        <label for="email" class="block text-sm uppercase">Email</label>
+
+        <input id="email" name="email" v-model="form.email" class="rounded-md px-2 py-1" required />
+      </div>
+
+      <div>
+        <label for="password" class="block text-sm uppercase">password</label>
+
+        <input
+            id="password"
+            name="password"
+            v-model="form.password"
+            class="rounded-md px-2 py-1"
+            type="password"
+            required
+        />
+      </div>
+
+      <div class="pt-2">
+        <button type="submit">Register</button>
+      </div>
+
+      <div v-if="error">{{ error }}</div>
+    </form>
+  </div>
+</template>
+
+```
+
+### 3. Create an API route for registration
+
+With the user interface created we'll need to add a route to receive the registration form data. This route should accept a POST request with the email and password in the request body. It should hash the password and insert the user into the database. This route will only accept POST requests, so we'll follow the instructions for [API route methods](https://nuxt.com/docs/guide/directory-structure/server#matching-http-method) and name the file with `*.post.ts` to restrict the endpoint to only accept POST.
```

Comment:
"file with `*.post.ts`" can likely be rephrased into "file `*.post.ts`", as `name the file with` doesn't quite fit (you name something some name, you don't name something with some name)

"to restrict the endpoint to only accept POST." seems a little convoluted and can likely be rephrased into "to restrict the endpoint to POST requests".